jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2013, 2015, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.zip;
+
+//import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public
+class ZipUtils {
+
+    /**
+     * J2ObjC added: avoids using java.nio.file.attribute.FileTime,
+     * instead implements FileTime class with relevant methods.
+     */
+    public static class FileTime implements Comparable<FileTime>{
+
+        private final long millis;
+
+        FileTime(long millis) {
+            this.millis = millis;
+        }
+
+        static FileTime from(long value, TimeUnit unit) {
+            if (unit.equals(TimeUnit.MICROSECONDS)) {
+                return new FileTime( value / 1000);
+            }
+            if (unit.equals(TimeUnit.MILLISECONDS)) {
+                return new FileTime(value);
+            }
+            if (unit.equals(TimeUnit.SECONDS)) {
+                return new FileTime(1000 * value);
+            }
+            return new FileTime(0);
```

Comment:
I feel that switch-case may be more appropriate here. How do you make sure that these are the only `TimeUnit` enums the package is using? Same question below.

Reply:
I changed the implementation into switch-case and included all TimeUnit enums.